packages/core/octo-linker.js

```diff
@@ -14,22 +14,6 @@ function initialize(self) {
 
   self._blobReader = new BlobReader();
   self._pluginManager = new Plugins(loadPlugins);
-  clickHandler(self._pluginManager);
-}
-
-function insertDataAttr(matches) {
-  matches.forEach(item => {
-    if (!item) {
-      return;
-    }
-
-    const { data, link } = item;
-    for (const key in data) {
-      if (data.hasOwnProperty(key)) {
-        link.dataset[key] = data[key];
-      }
-    }
-  });
```

Comment:
We used data-attributes to hold a reference to the plugin and other plugin related informations. Now we pass the plugin itself down which makes things easier.